timely_beliefs/tests/test_db_subclassing.py

```diff
@@ -82,9 +82,20 @@ def source_id(cls):
         "RatedSourceInCustomTable", backref=backref("beliefs", lazy=True)
     )
 
-    def __init__(self, sensor, source, happiness: float = None, **kwargs):
+    def __init__(
+        self,
+        sensor: DBSensor,
+        source: DBBeliefSource,
+        happiness: float = None,
+        **kwargs
+    ):
         self.happiness = happiness
         TimedBeliefDBMixin.__init__(self, sensor, source, **kwargs)
+        [
+            kwargs.pop(key, None)
```

Comment:
Is this line always need when creating a DB class based on a mixin? I'm not sure I get what this doing so please add a comment.

Reply:
This is a small experiment in giving developers maximum flexibility. One of the benefits of passing kwargs to a super class is that a developer can change the function signature of the super class without breaking sub classes. In this case there are 2 super classes, which actually have different keyword arguments. I'm assuming no arguments are named the same (we could check that actually), and take out any named arguments used by the first super class to avoid having the second super class complain about keyword arguments it doesn't recognize. This way, the benefit I mentioned works for both super classes.

That said, there are also [popular arguments](https://stackoverflow.com/questions/14626279/inheritance-best-practice-args-kwargs-or-explicitly-specifying-parameters) against passing kwargs to a super class, and for being explicit about the function parameters that each super class needs. I restricted this usage of kwargs to the test only, and for the actual implementation I went with explicit function parameters for maximum documentation. Note that the readme example still uses kwargs (for brevity), but all kwargs go the mixin (and none to `Base`).

I'll add a comment to the test as you suggested.